Survey/SurveyUnitTests/Sources/Specs/Login/LoginNetworkAPISpec.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  LoginNetworkAPISpec.swift
+//  SurveyUnitTests
+//
+//  Created by Taher on 14/11/22.
+//
+//  swiftlint:disable function_body_length closure_body_length
+
+import Combine
+import Nimble
+import Quick
+
+@testable import Survey
+
+final class LoginNetworkAPISpec: QuickSpec {
+
+    override func spec() {
+
+        var loginNetwork: LoginNetworkAPI!
+        var loginTokenData: LoginTokenApi?
+        var cancellables = CancelBag()
+        let baseUrl = URL(string: "https://survey-api.nimblehq.co/api/v1/oauth/token")
+
+        describe("a LoginNetworkAPI") {
+
+            beforeEach {
+                loginTokenData = nil
+            }
+
+            context("when a valid response is provided") {
+
+                beforeEach {
+                    guard let url = baseUrl else {
+                        return
+                    }
+                    let tokenData = LoginTokenApi.createDummy()
+                    var mockLoginData = Data()
+                    let encoder = JSONEncoder()
+                    do {
+                        mockLoginData = try encoder.encode(tokenData)
+                    } catch {
+                        NSLog(error.localizedDescription)
+                    }
```

Comment:
These steps to create mock data for testing seems busy here, could you make some kind of utility/helper for that 🙏 
Also, logging error (by means of `NSLog`) is necessary for production code, but for unit tests is not I think. Try to make use of `guard let ... = try?` instead.